build.sbt

```diff
@@ -48,6 +48,29 @@ def scaladocSourceUrl(dir: String) =
     )
   }
 
+def slickScalacOptions = Seq(
```

Comment:
I created this because the new `slickCollectionsCompat` module requires
`scalacOptions` from `slickGeneralSettings` but not the other settings because they are irrelevant.

One can make an argument that not actually not all of these `scalacOptions` are necessary for `slickCollectionsCompat`, in which case I can just directly copy the relevant `scalacOptions`  into `slickCollectionsCompat`, this having the bonus of reducing the diff for the other code.